internal/db/modelmigration_test.go

```diff
@@ -87,22 +101,94 @@ func (s *dbSuite) TestGetModelMigration(c *qt.C) {
 	c.Assert(eError.Code, qt.Equals, errors.CodeNotFound)
 }
 
-func (s *dbSuite) TestGetModelMigrationsCreatedBefore(c *qt.C) {
-	err := s.Database.Migrate(context.Background())
-	c.Assert(err, qt.Equals, nil)
+func (s *dbSuite) TestGetModelMigrationWithLock_NoWait(c *qt.C) {
+	controller := s.setupModelMigrationTest(c)
 
-	cloud := dbmodel.Cloud{
-		Name: "test-cloud",
+	migration := dbmodel.IncomingModelMigration{
+		ModelUUID:          sql.NullString{String: "00000001-0000-0000-0000-000000000001", Valid: true},
+		TargetControllerID: controller.ID,
+		UserMapping:        dbmodel.StringMap{"local": "external"},
 	}
-	err = s.Database.AddCloud(context.Background(), &cloud)
-	c.Assert(err, qt.IsNil)
+	c.Assert(s.Database.DB.Create(&migration).Error, qt.IsNil)
 
-	controller := dbmodel.Controller{
-		Name:      "test-controller",
-		UUID:      "00000000-0000-0000-0000-000000000001",
-		CloudName: cloud.Name,
+	notifyChan := make(chan struct{})
+	noWait := true
+	go func() {
+		// Lock the migration
+		err := s.Database.Transaction(func(d *db.Database) error {
+			lookup := dbmodel.IncomingModelMigration{
+				ModelUUID: migration.ModelUUID,
+			}
+			err := d.GetIncomingModelMigrationWithLock(context.Background(), &lookup, noWait)
+			c.Check(err, qt.Equals, nil)
```

Comment:
are we ok using Check in a goroutine?

Reply:
Yeah check() is okay as it will mark the test as failed but won't stop the running routine if it fails. Assert should be avoided in a go routine because if it fails the go routine will stop.